### PR TITLE
smtp set tls ciphers option

### PR DESCRIPTION
**Pull request by bernatvadell**
Yep!

I have tried to configure a gmail account as an email provider via SMTP transport. 

My initial config was:
```
      EMAIL_TRANSPORT: "smtp"
      EMAIL_SMTP_HOST: "smtp.gmail.com"
      EMAIL_SMTP_PORT: "587"
      EMAIL_SMTP_USER: "xxxxx"
      EMAIL_SMTP_PASSWORD: "xxxx"
      EMAIL_SMTP_SECURE: "true"
      EMAIL_SMTP_POOL: "false"
      EMAIL_SMTP_IGNORE_TLS: "false"
```

With this config, i've this issue:
```
{
  "message": "140102489255240:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:\n",
  "extensions": {
    "code": "INTERNAL_SERVER_ERROR"
  }
}
```

Googling i found that:
https://stackoverflow.com/questions/66317125/node-js-nodemailer-error-wrong-version-number-invalid-greeting

For this reason, I have created this PR, so that through environment variables, this configuration can be specified. 

Now my config like as:
```
     EMAIL_TRANSPORT: "smtp"
      EMAIL_SMTP_HOST: "smtp.gmail.com"
      EMAIL_SMTP_PORT: "587"
      EMAIL_SMTP_USER: "xxxx"
      EMAIL_SMTP_PASSWORD: "xxxx"
      EMAIL_SMTP_SECURE: "true"
      EMAIL_SMTP_POOL: "false"
      EMAIL_SMTP_IGNORE_TLS: "false"
      EMAIL_SMTP_TLS_CIPHERS: "SSLv3"
```

### CODE REVIEW

File: api/src/mailer.ts [hunk 1]
```diff
@@ -23,13 +23,20 @@ export default function getMailer(): Transporter {
 			};
 		}
 
+		const tls: Record<string, unknown> = {};
+
+		if (env.EMAIL_SMTP_TLS_CIPHERS) {
+			tls.ciphers = env.EMAIL_SMTP_TLS_CIPHERS;
+		}
+
```
Comment (rijkvanzanten) [inline api/src/mailer.ts:31]: We should rely on the getConfig utility here. That'll also ensure we support any and all other configuration options, and maintains consistency with other configuration properties (like storage adapters)

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: api/src/mailer.ts (hunk 1 above)
Comment (bernatvadell) [reply api/src/mailer.ts:31]: Can you give me an example of how to implement getConfig utility? Ty!

Comment (rijkvanzanten) [reply api/src/mailer.ts:31]: This is the utility: https://github.com/directus/directus/blob/main/api/src/utils/get-config-from-env.ts

https://github.com/directus/directus/blob/87ab9b3dfa3769e2b391619017f872cc5d3bd53a/api/src/middleware/session.ts#L11

You pass it the prefix of the options you want, and it'll return an object of the env vars (camelcased) that start with that prefix 🙂

Comment (bernatvadell) [reply api/src/mailer.ts]: done for tls config

**Comment by bernatvadell** [comment]
I have had a problem with the fork and it has been reset, I have had to recreate the PRs.

Follow #7082

**Pull request closed without merging**